File: airesources/CSharp/Halite2/hlt/Planet.cs
using System.Collections.Generic;

namespace Halite2.hlt {
    public class Planet: Entity {

        private int remainingProduction;
        private int currentProduction;
        private int dockingSpots;
        private IList<int> dockedShips;

        public Planet(int owner, int id, double xPos, double yPos, int health,
                      double radius, int dockingSpots, int currentProduction,
                      int remainingProduction, List<int> dockedShips)
        :base(owner, id, xPos, yPos, health, radius)
        {
            this.dockingSpots = dockingSpots;
            this.currentProduction = currentProduction;
            this.remainingProduction = remainingProduction;
            this.dockedShips = dockedShips.AsReadOnly();
        }

        public int RemainingProduction => remainingProduction;

        public int GetCurrentProduction() {
            return currentProduction;
        }

        public int GetDockingSpots() {
            return dockingSpots;
        }

        public IList<int> GetDockedShips() {
            return dockedShips;
        }

        public bool IsFull() {
            return dockedShips.Count == dockingSpots;
        }

        public bool IsOwned() {
            return GetOwner() != -1;
        }
        
        public override string ToString() {
            return "Planet[" +
                    base.ToString() +
                    ", remainingProduction=" + remainingProduction +
                    ", currentProduction=" + currentProduction +
                    ", dockingSpots=" + dockingSpots +
                    ", dockedShips=" + dockedShips +
                    "]";
        }
    }
}

File: airesources/Kotlin/src/kotlin/halite/Planet.kt
package halite

import java.util.Collections

class Planet(owner: Int, id: Int, xPos: Double, yPos: Double, health: Int,
             radius: Double, val dockingSpots: Int, val currentProduction: Int,
             val remainingProduction: Int, dockedShips: List<Int>) : Entity(owner, id, xPos, yPos, health, radius) {
    val dockedShips: List<Int> = Collections.unmodifiableList(dockedShips)

    val isFull: Boolean
        get() = dockedShips.size == dockingSpots

    val isOwned: Boolean
        get() = owner != -1

    override fun toString(): String {
        return "Planet[${super.toString()}, remainingProduction=$remainingProduction, currentProduction=$currentProduction, dockingSpots=$dockingSpots, dockedShips=$dockedShips]"
    }
}
